server/src/main/java/org/elasticsearch/rest/action/document/RestGetSourceAction.java

```diff
@@ -62,7 +70,13 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
-        final GetRequest getRequest = new GetRequest(request.param("index"), request.param("type"), request.param("id"));
+        final GetRequest getRequest;
+        if (request.hasParam("type")) {
+            deprecationLogger.deprecatedAndMaybeLog("get_source_type", TYPES_DEPRECATION_MESSAGE);
```

Comment:
Really small comment, but for consistency this would be `get_source_with_types`.

Reply:
I'm suprised this gets logges somewhere, so far I thought this only needs to be a unique key. But I'll change.